opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/services/FacadeService.java

```diff
@@ -0,0 +1,53 @@
+package de.adorsys.opba.protocol.facade.services;
+
+import de.adorsys.opba.db.domain.entity.ProtocolAction;
+import de.adorsys.opba.protocol.api.Action;
+import de.adorsys.opba.protocol.api.dto.context.ServiceContext;
+import de.adorsys.opba.protocol.api.dto.request.FacadeServiceableGetter;
+import de.adorsys.opba.protocol.api.dto.result.Result;
+import lombok.RequiredArgsConstructor;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+@RequiredArgsConstructor
+public abstract class FacadeService<I extends FacadeServiceableGetter, O, A extends Action<I, O>> {
+
+    private final ProtocolAction action;
+    private final Map<String, ? extends A> actionProviders;
+    private final ProtocolSelector selector;
+    private final ServiceContextProvider provider;
+    private final ResultHandler handler;
+
+    @Transactional
+    public CompletableFuture<Result<O>> execute(I request) {
```

Comment:
`final`?

Reply:
I would allow overriding it, final is too strict for me

Reply:
This is the core logic of loading protocol and execution delegation there. To me all deviations to be packed in the protocol layer. If we allow overriding here, then the whole logic can deviate, which may bring inconsistencies.

We can always provide some protected methods for overriding particular behaviour and delegate part of the main method there. To me it's more controllable way rather overriding the whole `execute` method here.

I don't force you though, I'm OK living here without `final`.